### PR TITLE
Use chunked parsing to parse Large XML files using xml_parse

### LINKED ISSUE #350: Parsing very large RDF/XML files fails
Parsing very large RDF/XML files fails, mainly because of problems with the *expat* based parser being used:
* https://www.php.net/manual/en/ref.xml.php

The error message seen is:
```
EasyRdf\Parser\Exception: XML error: "No memory" on line 125002, column 1
```

A number of Pull Requests have been raised that propose fixes for this:

- #291 - _"Allow for parsing of large XML files"_
- #294 - _"Implement LIBXML_PARSEHUGE"_
- #337 - _"RDFXML parser crashes any character sequences decoded by the urldecode()"_

EasyRdf was never designed for parsing/processing huge amounts of data but nevertheless, it really should be fixed.

I found the fixes in the Pull Requests quite hacky.  But after doing some tests I found that the [XML Reader](https://www.php.net/manual/en/book.xmlreader.php) interface, which uses libxml, is able to parse large strings without problems.

### PR DESCRIPTION
**Pull request by njh**
This is an alternative to #352. It fixes #350.

It works in the same ways as Example 1 in the PHP documentation:
https://www.php.net/manual/en/function.xml-parse.php

Pros:
* It is fast
* It works
* There is a test that passes
* This PR is complete

Cons:
* It doesn't use my shiny new [XMLParser](https://github.com/easyrdf/easyrdf/blob/master/lib/XMLParser.php) class

### CODE REVIEW

**Comment by njh** [comment]
Branch `xml-chunked-parsing`:

```
Input file: london.rdf
File size: 2840397 bytes
  Parse time: 352 ms
  Triple Count: 12861
  Resource count: 11799
  Total time: 409 ms
```

Branch `master`:
```
Input file: london.rdf
File size: 2840397 bytes
  Parse time: 353 ms
  Triple Count: 12861
  Resource count: 11799
  Total time: 411 ms
```

(Both test times fluctuate a bit when running a few times on my laptop, but I think what is important is that there is no significant performance difference)

**Comment by njh** [comment]
Just pushed an update that uses less memory.
It no-longer creates an array of chunks of the the input RDF/XML.
Instead it creates a single 1MB chunk at a time.

File: Makefile [hunk 1]
```diff
@@ -2,6 +2,7 @@ PACKAGE = easyrdf
 VERSION = $(shell php -r "print json_decode(file_get_contents('composer.json'))->version ?? 'dev';")
 distdir = $(PACKAGE)-$(VERSION)
 PHP = $(shell which php)
+PHP_FLAGS = -d memory_limit=1G
```
Comment (k00ni) [inline Makefile:5]: I was wondering what memory requirements your new approach has. Is 1 GB only for tests here?

**Review by k00ni — APPROVED** [review]
Everything looks fine, I only had one question (See below).

File: Makefile (hunk 1 above)
Comment (njh) [reply Makefile:5]: Yeah, the memory increase is to make sure there is enough memory for the "large RDF/XML file" test to run.

There is no increase in memory requirement, unless you want to load large files.

**Pull request merged**